src/Common/src/System/Security/Cryptography/DerEncoder.cs

```diff
@@ -324,5 +517,99 @@ internal static byte[] ConstructSequence(params byte[][][] items)
 
             return encodedSequence;
         }
+
+        private static BigInteger ParseOidRid(string oidValue, int startIndex, int endIndex)
+        {
+            if (startIndex > oidValue.Length)
+                throw new CryptographicException();
+
+            if (endIndex == -1)
+            {
+                endIndex = oidValue.Length;
```

Comment:
I updated this a bit, now it just takes `ref int startIndex` and hides its bounds checking.